persistent/Database/Persist/TH.hs

```diff
@@ -1720,6 +1722,30 @@ mkMigrate fun allDefs = do
         m <- [|migrate|]
         return $ NoBindS $ m `AppE` defsExp `AppE` u
 
+makePersistEntityDefExp :: MkPersistSettings -> EntityMap -> EntityDef -> Q Exp
+makePersistEntityDefExp mps entityMap entDef@EntityDef{..} =
+    [|EntityDef
+        entityHaskell
+        entityDB
+        $(liftAndFixKey entityMap entityId)
+        entityAttrs
+        $(fieldDefReferences mps entDef entityFields)
+        entityUniques
+        entityForeigns
+        entityDerives
+        entityExtra
+        entitySum
+        entityComments
+    |]
+
+fieldDefReferences :: MkPersistSettings -> EntityDef -> [FieldDef] -> Q Exp
+fieldDefReferences mps entDef fieldDefs = do
+    lookupValueName "persistFieldDef" >>= \case
```

Comment:
This feels a bit dicey. Why not use the name facility lie `'persistFieldDef` ? That gives you a qualified name that should work fine in the context without requiring it to be imported.

eg:

```haskell
fmap ListE $ forM fieldDefs $ \fieldDef -> do
  ...
    pure $ VarE 'persistFieldDef `AppE` fieldDefConE
```